provisioning/image_customization_service.go

```diff
@@ -0,0 +1,65 @@
+package provisioning
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+)
+
+const (
+	imageCustomizationService = "metal3-image-customization-service"
+	customizedImagePortName   = "customizedImagePortName"
+	imageCustomizationPort    = "6184"
+)
+
+func newImageCustomizationService(targetNamespace string) *corev1.Service {
+	port, _ := strconv.Atoi(imageCustomizationPort) // #nosec
+
+	ports := []corev1.ServicePort{
+		{
+			Name: customizedImagePortName,
```

Comment:
This name should be something like "http"

Currently this is failing with:

> Failed to create Service/metal3-image-customization-service -n openshift-machine-api: Service "metal3-image-customization-service" is invalid: spec.ports[0].name: Invalid value: "customizedImagePortName": a lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?'

Reply:
Yes, I see that error and posted a fix. Lets see what happens next.